### DISCUSSION TITLE
convert shp data

### DISCUSSION THREAD

**Discussion by patrickstar9808** [Project-OSRM/osrm-backend#6838] [Q&A]
How to convert shp data into data that osrm extract can parse
For example, xxx.osm. pbf

**Comment by nilsnolde** [comment]
There’s 2 things:

- data format: to create a PBF you use eg pyosmium
- data model: if your data is non-OSM you likely attributed the roads differently, so during the conversion process you’ll need to map your attributes to OSM concepts, most importantly the „highway“ tag

It’s a bit a process and you’ll need to know OSM quite well, the data format (sorted by type etc) as well as the data semantics.

**Comment by patrickstar9808** [comment]
> There’s 2 things:
> 
> * data format: to create a PBF you use eg pyosmium
> * data model: if your data is non-OSM you likely attributed the roads differently, so during the conversion process you’ll need to map your attributes to OSM concepts, most importantly the „highway“ tag
> 
> It’s a bit a process and you’ll need to know OSM quite well, the data format (sorted by type etc) as well as the data semantics.


Thank you for your reply. “highway” is really important information.
